src/core/constants.js

```diff
@@ -245,6 +245,7 @@ export const GLOBAL_EVENT_STATUS_MAP = {
   onUninstalled: UNINSTALLED,
   onEnabling: ENABLING,
   onDisabling: DISABLING,
+  onOperationCancelled: INSTALLED,
```

Comment:
Should this always be mapped to install?

What about this:
- Go to `about:addons`
- Go to Extensions
- Click *Disable* for one of the extensions
- Click *Remove* on that same extension
- Click *undo*

After clicking undo, the extension goes back to disabled in `about:addons`. What happens to the install button for that case?

Reply:
I either need https://github.com/mozilla/addons-frontend/issues/6143 or a new hard drive to test with `mozAddonManager` 🙁

Reply:
mmh right, that will set the "installed" state to the AMO button whereas the `about:addons` shows "disabled".

I am not sure what to do though:

1. the initial issue is a super rare edge case
2. the code to hook into those listeners is not super extendable
3. your STR are a even more super rare edge case 😛 

We cannot easily set the previous state because, if we follow your STR, the previous state is "UNINSTALLING", not "DISABLED". We could filter the events to get "stable" events only and retain the last one, but this seems to introduce a lot of complexity.

Reply:
> I either need #6143 or a new hard drive to test with `mozAddonManager` 🙁

no more excuse now 😸

Reply:
so... I came up with a new patch here, which depends on https://github.com/mozilla/addons-frontend/pull/6237